File: Top Interview 150/Python/Remove_Duplicates_From_Sorted_Array.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        i = 0
        while i < len(nums) - 1:
            if nums[i] == nums[i + 1]:
                nums.pop(i)
            else:
                i += 1
        return len(nums)

print(Solution().removeDuplicates([1, 1, 2]))  # 2
print(Solution().removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]))  # 5
print(Solution().removeDuplicates([1, 2]))  # 2
        

File: Top Interview 150/C++/Remove_Duplicates_From_Sorted_Array.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        short i = 0;
        while(i < nums.size() - 1){
            if(nums[i] == nums[i+1]) nums.erase(nums.begin() + i);
            else i++;
        }
        return nums.size();
    }
};

int main(){
    Solution s;
    vector<int> nums = {0,0,1,1,1,2,2,3,3,4};
    cout << s.removeDuplicates(nums) << endl; // 5
    for(int i = 0; i < nums.size(); i++){
        cout << nums[i] << " ";
    }
    cout << endl;
    return 0;
}